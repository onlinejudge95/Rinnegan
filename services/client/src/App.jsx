import React from "react";
import { Route, Switch } from "react-router-dom";
import axios from "axios";

import authComponents from "./components/auth";
import userComponents from "./components/user";
import navBarComponents from "./components/navbar";
import keyWordComponents from "./components/keyword";
import About from "./components/About";
import Message from "./components/Message";

class App extends React.Component {
  state = {
    user: null,
    accessToken: null,
    messageText: null,
    messageType: null,
  };

  onRegisterFormSubmit = async (payload) => {
    try {
      const registerApiUrl = `${process.env.REACT_APP_SERVER_URL}/auth/register`;
      const headers = {
        Accept: "application/json",
        "Content-Type": "application/json",
      };
      await axios.post(registerApiUrl, payload, { headers });
      this.createMessage("green", "You have registered successfully.");
    } catch (error) {
      this.createMessage("red", "That user already exists.");
      console.log(error);
    }
  };

  onLoginFormSubmit = async (payload) => {
    try {
      const loginApiUrl = `${process.env.REACT_APP_SERVER_URL}/auth/login`;
      let headers = {
        Accept: "application/json",
        "Content-Type": "application/json",
      };
      const authResponse = await axios.post(loginApiUrl, payload, { headers });
      this.createMessage("green", "Sign-In successful.");
      this.setState({ accessToken: authResponse.data.access_token });
      window.localStorage.setItem(
        "refreshToken",
        authResponse.data.refresh_token
      );

      const userDetailApiUrl = `${process.env.REACT_APP_SERVER_URL}/users/${authResponse.data.user_id}`;
      headers.Authorization = `Bearer ${this.state.accessToken}`;
      const userResponse = await axios.get(userDetailApiUrl, { headers });
      this.setState({ user: userResponse.data });
    } catch (error) {
      console.log("Invalid credentials");
    }
  };

  onUpdateUserFormSubmit = async (payload) => {
    try {
      const updateUserApiUrl = `${process.env.REACT_APP_SERVER_URL}/users/${this.state.user.id}`;
      const headers = {
        Accept: "application/json",
        Authorization: `Bearer ${this.state.accessToken}`,
        "Content-Type": "application/json",
      };
      delete payload.password;
      const userResponse = await axios.put(updateUserApiUrl, payload, {
        headers,
      });
      this.setState({ user: userResponse.data });
    } catch (error) {
      console.log("Invalid credentials");
    }
  };

  onRemoveUserClick = async () => {
    try {
      const deleteUserApiUrl = `${process.env.REACT_APP_SERVER_URL}/users/${this.state.user.id}`;
      const headers = {
        Accept: "application/json",
        Authorization: `Bearer ${this.state.accessToken}`,
      };
      await axios.delete(deleteUserApiUrl, {
        headers,
      });
      window.localStorage.removeItem("refreshToken");
      this.setState({ accessToken: null });
    } catch (error) {}
  };

  onLogOutClick = () => {
    window.localStorage.removeItem("refreshToken");
    this.setState({ accessToken: null });
  };

  isAuthenticated = () => {
    if (this.state.accessToken) {
      return true;
    }
    return false;
  };

  createMessage = (type, text) => {
    this.setState({
      messageText: text,
      messageType: type,
    });

    setTimeout(() => {
      this.removeMessage();
    }, 5000);
  };

  removeMessage = () => {
    this.setState({ messageText: null, messageType: null });
  };

  onAddKeywordFormSubmit = async () => {
    console.log("Hello");
  };

  render() {
    return (
      <div className="ui">
        <navBarComponents.NavBar
          isAuthenticated={this.isAuthenticated}
          handleLogOutClick={this.onLogOutClick}
        />
        <div className="ui container">
          <div className="ui segment">
            {this.state.messageType && this.state.messgeText && (
              <Message
                messageText={this.state.messageText}
                messageType={this.state.messageType}
                removeMessage={this.removeMessage}
              />
            )}
            <div className="ui grid">
              <div className="two wide column" />
              <div className="twelve wide column">
                <Switch>
                  <Route path="/" exact component={About} />
                  <Route
                    path="/register"
                    exact
                    component={() => (
                      <authComponents.RegisterUser
                        handleRegisterFormSubmit={this.onRegisterFormSubmit}
                        isAuthenticated={this.isAuthenticated}
                      />
                    )}
                  />
                  <Route
                    path="/login"
                    exact
                    component={() => (
                      <authComponents.LoginUser
                        handleLoginFormSubmit={this.onLoginFormSubmit}
                        isAuthenticated={this.isAuthenticated}
                      />
                    )}
                  />
                  <Route
                    path="/profile"
                    exact
                    component={() => (
                      <userComponents.ShowUser
                        user={this.state.user}
                        handleRemoveUserClick={this.onRemoveUserClick}
                        isAuthenticated={this.isAuthenticated}
                      />
                    )}
                  />
                  <Route
                    path="/update"
                    exact
                    component={() => (
                      <userComponents.UpdateUser
                        user={this.state.user}
                        isAuthenticated={this.isAuthenticated}
                        handleUpdateUserFormSubmit={this.onUpdateUserFormSubmit}
                      />
                    )}
                  />
                  <Route
                    path="/keyword"
                    exact
                    component={() => (
                      <keyWordComponents.KeywordList
                        isAuthenticated={this.isAuthenticated}
                      />
                    )}
                  />
                  <Route
                    path="/keyword/1"
                    exact
                    component={() => (
                      <keyWordComponents.KeywordDetail
                        isAuthenticated={this.isAuthenticated}
                      />
                    )}
                  />
                </Switch>
              </div>
              <div className="two wide column" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
