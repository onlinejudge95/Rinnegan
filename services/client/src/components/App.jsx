import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import RegisterUser from "./auth/RegisterUser";
import Loginuser from "./auth/LoginUser";
import ShowUser from "./user/ShowUser";
import RemoveUser from "./user/RemoveUser";
import UpdateUser from "./user/UpdateUser";
import About from "./About";
import NavBar from "./NavBar";

const App = () => {
  return (
    <div classNam="ui container">
      <BrowserRouter>
        <div>
          <NavBar />
          <Route path="/" exact component={About}></Route>
          <Route path="/register" exact component={RegisterUser}></Route>
          <Route path="/login" exact component={Loginuser}></Route>
          <Route path="/profile" exact component={ShowUser}></Route>
          <Route path="/update" exact component={UpdateUser}></Route>
          <Route path="/remove" exact component={RemoveUser}></Route>
        </div>
      </BrowserRouter>
    </div>
  );
};

export default App;
