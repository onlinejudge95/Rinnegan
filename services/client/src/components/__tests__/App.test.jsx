import React from "react";
import { render, cleanup } from "@testing-library/react";

import App from "../../App";

afterEach(cleanup);

it("renders", () => {
  const dom = renderWithRouter(<App />);
  expect(dom.asFragment()).toMatchSnapshot();
});
