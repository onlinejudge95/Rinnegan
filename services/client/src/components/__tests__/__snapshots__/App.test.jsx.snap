// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <nav
      aria-label="main navigation"
      class="navbar is-dark"
      role="navigation"
    >
      <section
        class="container"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item nav-title"
            href="/"
            style="font-weight: bold;"
          >
            Sentimental
          </a>
          <span
            class="nav-toggle navbar-burger"
          >
            <span />
            <span />
            <span />
          </span>
        </div>
        <div
          class="navbar-menu"
        >
          <div
            class="navbar-end"
          >
            <a
              class="navbar-item"
              data-testid="nav-register"
              href="/register"
            >
              Register
            </a>
            <a
              class="navbar-item"
              data-testid="nav-login"
              href="/login"
            >
              Log-In
            </a>
          </div>
        </div>
      </section>
    </nav>
    <section
      class="section"
    >
      <div
        class="container"
      >
        <div
          class="columns"
        >
          <div
            class="column is-half"
          >
            <br />
            <div>
              <h1
                class="title is-1"
              >
                Sentimental
              </h1>
              <hr />
              <br />
              <p
                class="content"
              >
                With the advent of Machine Learning, the filed of sentiment analysis has become somewhat trivial. You can find pre-made solutions to find sentiment of a word in a given text. We built upon this functionality and made this system.
              </p>
              <p
                class="content"
              >
                Sentimental is a platform where a user can come and serach for a 
                <strong>
                  sentiment analysis
                </strong>
                 on a given 
                <strong>
                  keyword
                </strong>
                . After he has finalized his choice, he will be shown the analysis for that word by choosing the corpus from
                <strong>
                  one or more of the social media platorms
                </strong>
                . Currently we support the following platforms to fetch data from
              </p>
              <div
                class="list is-hoverable"
              >
                <a
                  class="list-item"
                  href="https://twitter.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Twitter
                </a>
              </div>
              <p
                class="content"
              >
                As and when our infrastructure grows we will be using other platforms to include more diverse sentiments.
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
